tests/test_flask_admin.py

```diff
@@ -407,6 +408,162 @@ def test_browse_logs(self):
             self.assertTrue('<h1>Logs</h1>' in output.data)
             self.assertTrue('added the distro named: Debian' in output.data)
 
+    def test_browse_flags(self):
+        """ Test the browse_flags function. """
+
+        create_flagged_project(self.session)
+
+        output = self.app.get('/flags', follow_redirects=True)
+        self.assertEqual(output.status_code, 200)
+        self.assertTrue(
+            '<li class="errors">Login required</li>' in output.data)
+
+        with anitya.app.APP.test_client() as c:
+            with c.session_transaction() as sess:
+                sess['openid'] = 'openid_url'
+                sess['fullname'] = 'Pierre-Yves C.'
+                sess['nickname'] = 'pingou'
+                sess['email'] = 'pingou@pingoured.fr'
+
+            output = c.get('/flags', follow_redirects=True)
+            self.assertEqual(output.status_code, 401)
+
+        with anitya.app.APP.test_client() as c:
+            with c.session_transaction() as sess:
+                sess['openid'] = 'http://pingou.id.fedoraproject.org/'
+                sess['fullname'] = 'Pierre-Yves C.'
+                sess['nickname'] = 'pingou'
+                sess['email'] = 'pingou@pingoured.fr'
+
+            output = c.get('/flags')
+            self.assertEqual(output.status_code, 200)
+            self.assertTrue('<h1>Flags</h1>' in output.data)
+            self.assertTrue('geany' in output.data)
+
+            output = c.get('/flags?page=abc&limit=def&from_date=ghi')
+            self.assertEqual(output.status_code, 200)
+            self.assertTrue('<h1>Flags</h1>' in output.data)
+            self.assertTrue('geany' in output.data)
+
+            output = c.get('/flags?from_date=%s' % datetime.date.today())
```

Comment:
Just curious, if the from_date is tomorrow, then geany does not appear, correct?